src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023segmentationinterface.py

```diff
@@ -20,6 +22,59 @@ def __init__(
         end_frame: int,
         verbose: bool = True,
     ):
-        
-        super().__init__(folder_path=folder_path, channel_name=channel_name, plane_name=plane_name, plane_segmentation_name=plane_segmentation_name, verbose=verbose)
-        self.segmentation_extractor=self.segmentation_extractor.frame_slice(start_frame=start_frame,end_frame=end_frame)
+        super().__init__(
+            folder_path=folder_path,
+            channel_name=channel_name,
+            plane_name=plane_name,
+            plane_segmentation_name=plane_segmentation_name,
+            verbose=verbose,
+        )
+        self.segmentation_extractor = self.segmentation_extractor.frame_slice(
+            start_frame=start_frame, end_frame=end_frame
+        )
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        device_name = "CustomMicroscope"  # TODO how can I use the yaml file spec here?
```

Comment:
If you're updating the metadata with the editable metadata (yaml) in the `convert_session()` script, then I think here at `metadata = super().get_metadata()` you should see two devices, the default device and the "CustomMicroscope"?

Reply:
Then I am doing something wrong because I see only the default device